From: Dan Williams <dan.j.williams@intel.com>
Date: Sat, 1 May 2010 18:14:57 -0700
Subject: [PATCH] md: notify mdstat waiters of level change
Git-commit: bb7f8d2217d8753ab5008c78f16697d9e697d570
Patch-mainline: v2.6.35-rc1

Level modifications change the output of mdstat.  The mdmon manager
thread is interested in these events for external metadata management.

Signed-off-by: Dan Williams <dan.j.williams@intel.com>
Acked-by: NeilBrown <neilb@suse.de>
---
 drivers/md/md.c |    1 +
 1 file changed, 1 insertion(+)

--- linux-2.6.32-SLE11-SP1.orig/drivers/md/md.c
+++ linux-2.6.32-SLE11-SP1/drivers/md/md.c
@@ -3057,6 +3057,7 @@ level_store(mddev_t *mddev, const char *
 	set_bit(MD_RECOVERY_NEEDED, &mddev->recovery);
 	md_wakeup_thread(mddev->thread);
 	sysfs_notify(&mddev->kobj, NULL, "level");
+	md_new_event(mddev);
 	return rv;
 }
 
